product_cost_usd/models/product_template.py

```diff
@@ -0,0 +1,42 @@
+from odoo.tools import float_compare
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    standard_price_usd = fields.Float(
+        "Cost in USD", digits="Product Price", help="Price cost of the product in USD currency"
+    )
+
+    @api.constrains("standard_price_usd", "seller_ids")
+    def check_cost_and_price(self):
+        """Validate 'Cost in USD' usability.
+
+        Usability conditions:
+        - Before set a 'Cost in USD' in a product at least one supplier should
+          have price in USD.
+        - The Cost in USD cannot be less than supplier price.
+        """
+        usd_currency = self.env.ref("base.USD")
+        prec = self.env["decimal.precision"].precision_get("Product Price")
+        usd_seller = self.seller_ids.filtered(lambda x: x.currency_id == usd_currency)
+        list_price = usd_seller.price if usd_seller else 0.0
```

Comment:
If seller is not found, it will be an empty recordset, and `empty.float_field` isalready 0.0:

```suggestion
        list_price = usd_seller.price
```